runtimevar/paramstore/paramstore.go

```diff
@@ -33,32 +33,21 @@ import (
 	"github.com/aws/aws-sdk-go/service/ssm"
 )
 
-// Client stores long-lived variables for connecting to Parameter Store.
-type Client struct {
-	sess client.ConfigProvider
-}
-
-// NewClient returns a constructed Client with the required values.
-func NewClient(p client.ConfigProvider) *Client {
-	return &Client{sess: p}
-}
-
-// NewVariable constructs a runtimevar.Variable object with this package as the driver
-// implementation.
-func (c *Client) NewVariable(name string, decoder *runtimevar.Decoder, opts *Options) (*runtimevar.Variable, error) {
-	w, err := c.newWatcher(name, decoder, opts)
+// NewVariable constructs a runtimevar.Variable that watches AWS Parameter Store.
+func NewVariable(name string, sess client.ConfigProvider, decoder *runtimevar.Decoder, opts *Options) (*runtimevar.Variable, error) {
```

Comment:
`sess` usually comes earlier in the argument list, since it's a general credential-ish thing.

```suggestion
func NewVariable(sess client.ConfigProvider, name string, decoder *runtimevar.Decoder, opts *Options) (*runtimevar.Variable, error) {
```

Reply:
I'm going to open a bug to discuss, because we have the same thing in `blob`; e.g.:

```
func OpenBucket(ctx context.Context, bucketName string, sess client.ConfigProvider, opts *Options) (*blob.Bucket, error) 
```